File: swift/long_hair.swift
var ans : Dictionary<String, Int> = Dictionary<String, Int>()
let times = 5;
for (var cnt = 0; cnt < times; cnt++) {
    let key = readLine()!
    if (ans[key] == nil) {
        ans[key] = 0
    }
    ans[key]! += 1;
}
var maxKey :String?
var max = 0;
for (key, val) in ans {
    if (val > max) {
        max = val;
        maxKey = key;
    }
}
if (maxKey != nil) {
    print(maxKey!)
}

File: php/long_hair.php
<?php
    $ans = array();
    const TIMES = 5;
    for ($cnt = 0; $cnt < TIMES; $cnt++) {
        $key = trim(fgets(STDIN));
        if (!$ans[$key]) {
            $ans[$key] = 0;
        }
        $ans[$key] += 1;
    }
    $maxKey = "";
    $maxVal = 0;
    foreach ($ans as $key => $val) {
        if ($val > $maxVal) {
            $maxVal = $val;
            $maxKey = $key;
        }
    }
    echo $maxKey . "\n";
?>
